File: Semester 4/Programming and Projection Enironments/mpp-proiect-java/Proiect_MPP/AirportServices/src/main/java/root/services/IService.java
package root.services;

import root.model.Airport;
import root.model.Flight;
import root.model.people.Client;
import root.model.people.Employee;
import root.model.people.Person;

import java.time.LocalDate;
import java.util.Collection;
import java.util.List;

public interface IService {
    // Employee findEmployeeByEmail(String email) throws Exception;
    Employee login(Employee employee, IObserver client) throws Exception;
    void logout(Employee employee, IObserver client) throws Exception;

    Collection<Flight> findFlightByDestinationDate(String destination, LocalDate date) throws Exception;
    Airport findAirportById(int id) throws Exception;
    Flight findFlightById(int id) throws Exception;
    Collection<Flight> getAllAvailableFlights() throws Exception;
    Collection<Airport> getAllAirports() throws Exception;

    //void updateFlight(int id, int numberOfSeats) throws Exception;

    void buyTicket(Client client, List<Person> people, Flight flight) throws Exception;
}

File: Semester 4/Programming and Projection Enironments/mpp-proiect-csharp/Proiect_MPP/services/IService.cs
using model.people;
using model;
using System;
using System.Collections.Generic;
using System.Collections.ObjectModel;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace services
{
    public interface IService
    {
        Employee login(Employee employee, IObserver client);
        void logout(Employee employee, IObserver client);

        List<Flight> findFlightByDestinationDate(string destination, DateTime date);
        Airport findAirportById(int id);
        Flight findFlightById(int id);
        List<Flight> getAllAvailableFlights();
        List<Airport> getAllAirports();

        void buyTicket(Client client, List<Person> people, Flight flight);
    }
}
